### ISSUE TITLE
Generated ID from `useId` is no valid CSS selector

### ISSUE THREAD

**Issue by wokalek** [nuxt/nuxt#25949]
### Environment

- Operating System: Linux
- Node Version:     v20.11.1
- Nuxt Version:     3.10.3
- CLI Version:      3.10.1
- Nitro Version:    2.8.1
- Package Manager:  pnpm@8.15.4
- Builder:          -
- User Config:      telemetry, devtools, srcDir, app, extends, modules, vite, appConfig, runtimeConfig, eslint, tailwindcss, css, postcss, colorMode, device, content, pwa, robots, sitemap, site
- Runtime Modules:  @nuxtjs/eslint-module@4.1.0, @nuxtjs/tailwindcss@6.11.4, @nuxtjs/robots@3.0.0, @nuxtjs/sitemap@5.1.0, @nuxtjs/color-mode@3.3.2, @nuxtjs/fontaine@0.4.1, @nuxtjs/device@3.1.1, @nuxt/content@2.12.0, @nuxt/image@1.3.0, @vueuse/nuxt@10.8.0, @vite-pwa/nuxt@0.5.0, nuxt-typed-router@3.6.1, nuxt-svgo@4.0.0, nuxt-schema-org@3.3.4, nuxt-lodash@2.5.3
- Build Modules:    -

### Reproduction

use `useId`, then try for querySelectorAll

### Describe the bug

I get this error when I use querySelectorAll

```
Uncaught DOMException: Failed to execute 'querySelectorAll' on 'Document': 'JfvpJe1itr:5' is not a valid selector.
```

But `document.getElementById('JfvpJe1itr:18')` is working.

Google suggests that this is [not a valid selector](https://stackoverflow.com/a/63179238/7274005).

Would it be better to use `_` insted of `:`?

**Comment by manniL** [comment]
Try `document.getElementById('JfvpJe1itr\:18')`

**Comment by wokalek** [comment]
> Try `document.getElementById('JfvpJe1itr\:18')`

Yes, I already found that you can escape with `\\`, like this 

```js
document.getElementById('JfvpJe1itr\\:18')
```

**Comment by wokalek** [comment]
I just think that since it was meant to conveniently use `useId()` for linking inside libraries, such as PhotoSwipe, which need to pass `id`, or they themselves take the id of the element from the attributes (and you have no control over this, internally they can use querySelectorAll), then it would be best to make the row generation as universal as possible.

**Comment by wokalek** [comment]
I just want to say that the `useId` method generates an invalid id for use in the id html attribute. querySelectorAll `test:hover`.

querySelectorAll implies that in the `#test:filter` selector the element `<div id="test:filter">` has the `:filter` part - the filter condition.

So... I think this is not good for usage.

**Comment by manniL** [comment]
> I just want to say that the `useId` method generates an invalid id for use in the id html attribute. 

This is not correct. It is a valid value for `id` (which allows everything except whitespaces) but simply **no valid CSS selector**.

> querySelectorAll implies that in the `#test:filter` selector the element `<div id="test:filter">` has the `:filter` part - the filter condition.

If you pass id's to a library that could use them as a selector, you should escape them then 👍

**Comment by danielroe** [comment]
Yes, let's fix it so it's a valid selector as well as id 🙏 Should be straightforward.

**Comment by selfire1** [comment]
May I suggest for this issue to be reopened? With https://github.com/nuxt/nuxt/pull/28285 Nuxt uses Vue's native `useID` which is great! But if I'm not mistaken Nuxt's "port" of Vue's `useID` doesn't include the update to make it a valid selector: https://github.com/vuejs/core/issues/11828

Happy to create another issue as well if that is easier 🙏

**Comment by wokalek** [comment]
@selfire1 vue implementation of useId implies using "v" prefix before generating id. Do you have an error about invalid selector or do you think there might be one?

If generation starts with "v" with default settings, then it is definitely a valid selector.

https://github.com/vuejs/core/blob/babfb4cbcbf98601d76c1d7653eae8d250ce2710/packages/runtime-core/src/helpers/useId.ts#L7-L10

**Comment by selfire1** [comment]
The generation starts with `v`, but it includes a `:`: `v:0-7-3-0-0-0-0-0`.

This was fixed upstream in Vue with this commit: https://github.com/vuejs/core/commit/babfb4cbcbf98601d76c1d7653eae8d250ce2710. That fix was shipped in Vue 3.5.3: https://github.com/vuejs/core/blob/main/CHANGELOG.md#353-2024-09-06. But the latest Vue version in Nuxt is `3.5.0`: https://github.com/nuxt/nuxt/pull/28285/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R102.

**Comment by danielroe** [comment]
Nuxt does not pin your Vue version. Just refresh your lockfile to upgrade Vue.
